i18n-msg.html

```diff
@@ -64,166 +65,202 @@
-
-      var xhr = new XMLHttpRequest();
-      xhr.open('GET', url);
-      xhr.onload = function(e) {
-        if (e.target.status != 200) {
-          return;
-        }
-
-        var resp = JSON.parse(e.target.response);
-        if (!this.msgid in resp) {
-          console.warn(this.localName + ': "' + this.msgid + '" message id was not found in ' + url);
-          return;
-        }
-
-        this.locales[this.language] = resp; // cache this locale.
-
-        this.notifyInstances();
-
-        fetching_ = false;
-
-        this.fire('i18n-language-ready', {langauge: this.language});
-      }.bind(this);
-      xhr.send();
-    },
-
-    updateElementMessage: function(opt_instance) {
-      var instance = opt_instance || this;
-
-      var msg = instance.locales[instance.language][instance.msgid];
-      if (msg && msg.message) {
-        instance.innerHTML = msg.message;
-      }
-    },
-
-    notifyInstances: function() {
-      for (var i = 0, instance; instance = instances[i]; ++i) {
-        instance.language = I18nMsg.lang;
-
-        if (!instance.locales[instance.language][instance.msgid]) {
-          console.warn(this.localName + ': "' + instance.msgid + '" message id was not found');
-          continue;
-        }
-
-        this.updateElementMessage(instance);
-      }
-    },
-
-    /**
-     * Returns a message in the current locale (set by `window.I18nMsg.lang`).
-     * @method getMsg
-     * @param {string=} opt_msgId Optional message id to lookup. If left off,
-     * the instance's `msgid` property is used.
-     * @return {string|null} Translated message or `null` if not found.
-     */
-    getMsg: function(opt_msgId) {
-      var msgId = opt_msgId || this.msgid;
-      var lang = this.locales[this.language];
-      if (lang && lang[msgId]) {
-        return lang[msgId].message;
-      }
-      return null;
-    }
-
-  });
-
-})();
+    (function () {
+
+        window.I18nMsg = window.I18nMsg || {
+            lang: null,
+            url: 'locales',
+            locales: {}
+        };
+
+        var instances = [];
+
+        var fetching_ = false;
+        /* If if there's an outstanding XHR fetching a locale file. */
+
+        var numInstancesUpdated_ = 0;
+
+        Polymer({
+            is: 'i18n-msg',
+
+            properties: {
+                /**
+                 * The message id (key) for this message.
+                 *
+                 * @attribute msgid
+                 * @type string
+                 * @default null
+                 */
+                msgid: {
+                    type: String,
+                    value: null
+                },
+
+                /**
+                 * The set language being used.
+                 *
+                 * @property language
+                 * @type string
+                 * @default null
+                 * @readonly
+                 */
+                language: {
+                    type: String,
+                    value: null,
```

Comment:
`readOnly: true` and you'll need to use the special method `this._setLanguage(val)` when setting it.